### PR TITLE
response body convenience methods

### LINKED ISSUE #7: convenience methods on response
have a look at what reqwest offers. At the very least we need to offer:

The receiver of the methods should be `async_web_client::ResponseBody`

Probably in this order:
- `async fn bytes(&self) -> Result<Vec<u8>>`
- `async fn string(&self) -> Result<String>`
- `async fn json<T: DeserializeOwned>(self) -> Result<T>`

### PR DESCRIPTION
**Pull request by sin-sch**
- add function for read bytes
- convert bytes into string
- convert string in json
- edit post example: now using the new json method

fixes #7

### CODE REVIEW

**Comment by FlorianUekermann** [comment]
fixes #7

File: src/http/mod.rs [hunk 1]
```diff
@@ -149,6 +153,24 @@ impl ResponseBody {
         self.inner.into_inner()
     }
 }
+impl ResponseBody {
+    pub async fn bytes(&mut self) -> Result<Vec<u8>, io::Error> {
```
Comment (FlorianUekermann) [inline src/http/mod.rs]: Let's add `limit: Option<usize>` to this and the other two methods using `take()`.
Implementation can be mostly copied from `body_vec` a few lines above.

All methods in ResponseExt can be removed.

File: src/http/mod.rs [hunk 2]
```diff
@@ -149,6 +153,24 @@ impl ResponseBody {
...
+        let mut result = Vec::new();
+        self.read_to_end(&mut result).await?;
+        Ok(result)
+    }
+
+    pub async fn string(&mut self) -> Result<String, io::Error> {
+        let bytes = self.bytes().await?;
```
Comment (FlorianUekermann) [inline src/http/mod.rs]: Implement directly using `read_to_string`. That should be more efficient by catching encoding issues without reading the whole body first.

**Review by FlorianUekermann — CHANGES_REQUESTED** [review]

File: src/http/mod.rs (hunk 2 above)
Comment (sin-sch) [reply src/http/mod.rs]: We may not need the `bytes` method then. We do not use it in the `string` anymore, if we use `read_to_string` instead.

Comment (FlorianUekermann) [reply src/http/mod.rs]: `bytes` is intended to be used directly by users as well

File: src/http/mod.rs [hunk 3]
```diff
@@ -1,16 +1,20 @@
+//use async_net::TcpStream;
```
Comment (FlorianUekermann) [inline src/http/mod.rs]: leftover, there are a couple of others like this

File: Cargo.toml [hunk 4]
```diff
@@ -19,6 +19,11 @@ futures-rustls = { version = "0.26", default-features = false }
 async-ws = { version = "0.4.0", optional = true }
 webpki-roots = "0.26.0"
 
+serde = { version = "1.0", features = ["derive"] }
+serde_json = "1.0"
+
+reqwest = "0.12.7"
```
Comment (FlorianUekermann) [inline Cargo.toml]: accidental dependency?

File: Cargo.toml [hunk 5]
```diff
@@ -19,6 +19,11 @@ futures-rustls = { version = "0.26", default-features = false }
 async-ws = { version = "0.4.0", optional = true }
 webpki-roots = "0.26.0"
```
Comment (FlorianUekermann) [inline Cargo.toml]: remove empty line

File: Cargo.toml [hunk 6]
```diff
@@ -19,6 +19,11 @@ futures-rustls = { version = "0.26", default-features = false }
...
 
+serde = { version = "1.0", features = ["derive"] }
+serde_json = "1.0"
```
Comment (FlorianUekermann) [inline Cargo.toml]: let's make these optional and add a `json` feature flag, which is enabled by default, see chat message from Nov 10 for details

**Review by FlorianUekermann — CHANGES_REQUESTED** [review]

**Pull request merged**